lib/storage/storage-s3.js

```diff
@@ -56,6 +70,26 @@ class StorageS3 extends StorageBase {
             // Anonymize only client IP
             ip = `${anonymizeIp(ip.substring(0, commaIndex))}${ip.substring(commaIndex, ip.length)}`;
         }
+        const sources = this.getSources(JSON.parse(item.config).content);
+        const metadata = {
```

Comment:
I had to think a bit about this. All this stuff is already "in" the metadata, right? Here you're taking information encoded in the configuration and re-encoding it as metadata.

One could do all this work in the metadata handler too. Then it's not something that needs to be done in the storage itself: it would work for file storage and s3 storage equally as well.

It would also mean if we change the metadata generation, old links would immediately benefit, instead of having to re-save the metadata.

BUT.... we would need to fetch the S3 bucket and parse in the metadata handler. But I think we have to do that anyway... thoughts?

Reply:
This information is already stored, but not readily available

The main benefit of this approach over the one you mention, is that we would not need to worry about old config versions when checking if there're editor configs present, as those old versions would already be parsed, we would only need to ensure that the new version works.

For the record, I do think that approach is better, but I worry about versioning there, as the config itself is not versioned (http://golden-layout.com/docs/Config.html)